brave/brave-core
# Remove window._brave_solana

Resolves https://github.com/brave/brave-browser/issues/25645

1. We now use solana-web3.js minified js exported solanaWeb3 to generate `PublicKey` and `Transaction`.
2. Script resources is packed in `IDR_BRAVE_WALLET_SOLANA_WEB3_JS`
3. `//brave/components/brave_wallet/resources/solana_provider_internal.js` is completely removed so iOS needs to load the script through resource in a similar way as we load it in v8 and access `solanaWeb3.PublicKey` and `solanaWeb3.Transaction` directly.

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Setup wallet 
2. Open any sites and open dev console
3. Type `window._brave_solana` should get `undefined`
4. Navigate to https://codesandbox.io/s/github/darkdh/solana-provider-test
5. Should be able to `Connect`
6. After connected, open dev console and type` window._brave_solana` should still get `undefined`
7. Test `SignTransaction` and `SignAllTransactions` should be successful
